pkg/sql/schemachanger/scop/mutation.go

```diff
@@ -530,6 +530,13 @@ type RemoveDatabaseRoleSettings struct {
 	DatabaseID descpb.ID
 }
 
+// RemoveUserPrivileges is used to revoke a user's privileges.
+type RemoveUserPrivileges struct {
+	mutationOp
+	DescID descpb.ID
+	User   string
+}
```

Comment:
User privileges are implicitly removed when dropping something, so you can remove this op altogether.

Reply:
I believe `DROP OWNED BY` should revoke user privileges for the role(s) on objects that aren't owned, so we need this op to drop those privileges without dropping the entire objects.

Reply:
Good catch. @jasonmchan see rules in `pkg/sql/schemachanger/scplan/internal/rules/op_drop.go`

Reply:
I think [this rule](https://github.com/cockroachdb/cockroach/blob/2461c0883b2440997ed48130b183ba371f0490ef/pkg/sql/schemachanger/scplan/internal/rules/op_drop.go#L202-L239) should handle this, since user privileges are part of the descriptor.

Reply:
Amazing. Good find.

Reply:
Excellent!